stdlib/fileutils/0/fileutils.rbs

```diff
@@ -0,0 +1,569 @@
+#     FileUtils.install(src, dest, **options)
+#     FileUtils.chmod(mode, list, **options)
+#     FileUtils.chmod_R(mode, list, **options)
+#     FileUtils.chown(user, group, list, **options)
+#     FileUtils.chown_R(user, group, list, **options)
+#     FileUtils.touch(list, **options)
+#
+# Possible `options` are:
+#
+# `:force`
+# :   forced operation (rewrite files if exist, remove directories if not empty,
+#     etc.);
+# `:verbose`
+# :   print command to be run, in bash syntax, before performing it;
+# `:preserve`
+# :   preserve object's group, user and modification time on copying;
+# `:noop`
+# :   no changes are made (usable in combination with `:verbose` which will
+#     print the command to run)
+#
+#
+# Each method documents the options that it honours. See also ::commands,
+# ::options and ::options_of methods to introspect which command have which
+# options.
+#
+# All methods that have the concept of a "source" file or directory can take
+# either one file or a list of files in that argument.  See the method
+# documentation for examples.
+#
+# There are some `low level' methods, which do not accept keyword arguments:
+#
+#     FileUtils.copy_entry(src, dest, preserve = false, dereference_root = false, remove_destination = false)
+#     FileUtils.copy_file(src, dest, preserve = false, dereference = true)
+#     FileUtils.copy_stream(srcstream, deststream)
+#     FileUtils.remove_entry(path, force = false)
+#     FileUtils.remove_entry_secure(path, force = false)
+#     FileUtils.remove_file(path, force = false)
+#     FileUtils.compare_file(path_a, path_b)
+#     FileUtils.compare_stream(stream_a, stream_b)
+#     FileUtils.uptodate?(file, cmp_list)
+#
+# ## module FileUtils::Verbose
+#
+# This module has all methods of FileUtils module, but it outputs messages
+# before acting.  This equates to passing the `:verbose` flag to methods in
+# FileUtils.
+#
+# ## module FileUtils::NoWrite
+#
+# This module has all methods of FileUtils module, but never changes
+# files/directories.  This equates to passing the `:noop` flag to methods in
+# FileUtils.
+#
+# ## module FileUtils::DryRun
+#
+# This module has all methods of FileUtils module, but never changes
+# files/directories.  This equates to passing the `:noop` and `:verbose` flags
+# to methods in FileUtils.
+module FileUtils
+  VERSION: String
+
+  type mode = Integer | String
+
+  type pathlist = String | Array[String]
+
+  # Changes the current directory to the directory `dir`.
+  #
+  # If this method is called with block, resumes to the previous working directory
+  # after the block execution has finished.
+  #
+  #     FileUtils.cd('/')  # change directory
+  #
+  #     FileUtils.cd('/', verbose: true)   # change directory and report it
+  #
+  #     FileUtils.cd('/') do  # change directory
+  #       # ...               # do something
+  #     end                   # return to original directory
+  #
+  def self.cd: (String dir, ?verbose: boolish) -> void
```

Comment:
Most (or all?) FileUtils methods are `module_function`. So let use `self?` instead of `self`.

Reply:
Thanks! I'll improve the signatures and tests! 💪

Reply:
BTW, it might be nice if we could write it as follows... 🤔 

```diff
-alias self.chdir self.cd
-alias chdir cd
+alias self?.chdir self?.cd
```

Reply:
Fixed via ca1d180

Reply:
> BTW, it might be nice if we could write it as follows... thinking



True. I heard the unexistence of `self?` alias is intentional by @soutaro because of rare.

But maybe we can add `self?` alias now. Previously I didn't know there are many singleton-instance aliases like that, so I thought `self?` is unnecessary. But we found demands for singleton-instance aliases.

Reply:
> we found demands for singleton-instance aliases.

Yes, I think so. We could reduce duplications in a case like `FileUtils`.

Reply:
@ybiquitous Thanks! I don't think adding new syntax for aliasing module functions makes much sense for now. Let me know if you find this is a frequently seen pattern.

Reply:
OK, thanks for your comment 👍